### PR TITLE
Test using the latest version of Microsoft.NETCore.App

### LINKED ISSUE #14501: Not flowing dependency updates for Microsoft.NETCore.App
This repo is only flowing dependency updates for packages, but not for the actually .NET Core runtime (Microsoft.NETCore.App). That means bot-generated PRs such as https://github.com/aspnet/EntityFrameworkCore/pull/14500 are not actually testing EFCore with the latest .NET Core.

The repo is currently using the runtime version that is baked into the .NET Core SDK. The SDK version is not automatically updated by Arcade, and even if it were, it would often be at least 1 step behind the latest runtime version which will be inserted into the SDK.

@bricelam @ajcvickers - was this an intentional change when converting this repo to Arcade?

### PR DESCRIPTION
**Pull request by bricelam**
Resolves #14501 (again)

### CODE REVIEW

File: Directory.Build.targets [hunk 1]
```diff
@@ -6,6 +6,22 @@
...
 
+  <ItemGroup>
+    <!--
+      Workaround the cyclical nature of building .NET Core itself.
+      The SDK implies a default version of .NET Core to use, but that is typically too stale for this repo.
+      This sets the version of the NETCore.App shared framework which should be used
+    -->
+    <KnownFrameworkReference Update="Microsoft.NETCore.App">
+      <!-- Always update the 'latest version', whether the repo is servicing or not. -->
+      <LatestRuntimeFrameworkVersion Condition="'%(TargetFramework)' == '$(DefaultNetCoreTargetFramework)'">$(MicrosoftNETCoreAppPackageVersion)</LatestRuntimeFrameworkVersion>
```
Comment (natemcmaster) [inline Directory.Build.targets]: Does this repo define DefaultNetCoreTargetFramework and IsServicingBuild? I didn't see them in my GitHub code search. Those are custom properties we set in the aspnet/Extensions repo 

https://github.com/aspnet/Extensions/blob/714b1bea552187a3716c0be3e04bd3830c05e60f/Directory.Build.props#L35
https://github.com/aspnet/Extensions/blob/42f0f5d278f4789fb26ccee5c5af5dbdaaea15b1/version.props#L13

Comment (bricelam) [reply Directory.Build.targets]: Thanks. The line between customizations and workarounds is a bit blurry. 😉 Will fix.

Comment (bricelam) [reply Directory.Build.targets]: Curios: Why is `TargetFramework` an item group here and not a property?

Comment (bricelam) [reply Directory.Build.targets]: Oh I get it. It's a filter on the current item set.

**Comment by bricelam** [comment]
And errors like this is why I ripped all this out:
```
error NU1605: Detected package downgrade: runtime.win-x64.Microsoft.NETCore.DotNetAppHost from 3.0.0-preview6-27706-03 to 3.0.0-preview4-27529-09. Reference the package directly from the project to select a different version.
error NU1605:  dotnet-ef -> runtime.win-x64.Microsoft.NETCore.DotNetHostResolver 3.0.0-preview6-27706-03 -> Microsoft.NETCore.DotNetAppHost 3.0.0-preview6-27706-03 -> runtime.win-x64.Microsoft.NETCore.DotNetAppHost (>= 3.0.0-preview6-27706-03)
error NU1605:  dotnet-ef -> runtime.win-x64.Microsoft.NETCore.DotNetAppHost (>= 3.0.0-preview4-27529-09)
```

**Comment by natemcmaster** [comment]
Hmm, does this work if you update global.json to use the preview5 SDK? (3.0.100-preview5-011568)

**Pull request merged**